modules/gapi/test/streaming/gapi_streaming_tests.cpp

```diff
@@ -1819,82 +1825,278 @@ TEST(GAPI_Streaming, Reshape)
+                                                         TestSourceType,
+                                                         std::function<cv::Mat(const cv::Mat&)>>
+                                                                >
+{ };
+
+TEST_P(GAPI_Accessors_Meta_In_Streaming, AccuracyTest)
 {
-    initTestDataPath();
-    std::string filepath = findDataFile("cv/video/768x576.avi");
+    std::string filepath { };
+    std::function<cv::GMat(const cv::GFrame&)> accessor;
+    TestSourceType sourceType;
+    std::function<cv::Mat(const cv::Mat&)> fromBGR;
 
-    cv::GFrame in;
-    auto out = cv::gapi::streaming::BGR(in);
+    std::tie(filepath, accessor, sourceType, fromBGR) = GetParam();
 
-    cv::GComputation comp(cv::GIn(in), cv::GOut(out));
+    initTestDataPathOrSkip();
+    const std::string& absFilePath = findDataFile(filepath, false);
+
+    cv::GFrame in;
+    cv::GMat gmat = accessor(in);
+    cv::GMat resized = cv::gapi::resize(gmat, cv::Size(1920, 1080));
+    cv::GOpaque<int64_t> outId = cv::gapi::streaming::seq_id(resized);
+    cv::GOpaque<int64_t> outTs = cv::gapi::streaming::timestamp(resized);
+    cv::GComputation comp(cv::GIn(in), cv::GOut(resized, outId, outTs));
 
     auto cc = comp.compileStreaming();
-    try {
-        cc.setSource<NV12Source>(filepath);
-    } catch(...) {
-        throw SkipTestException("Video file can not be opened");
-    }
+    auto src = createTestSource(sourceType, absFilePath);
+    cc.setSource(src);
 
     cv::VideoCapture cap;
-    cap.open(filepath);
+    cap.open(absFilePath);
     if (!cap.isOpened())
         throw SkipTestException("Video file can not be opened");
 
-    cv::Mat ocv_mat, gapi_mat;
+    cv::Mat cap_mat, req_mat, ocv_mat, gapi_mat;
+    int64_t seq_id, timestamp;
+    std::vector<int64_t> all_seq_ids;
+    std::vector<int64_t> all_timestamps;
+
     std::size_t num_frames = 0u;
     std::size_t max_frames = 10u;
 
     cc.start();
-    while (cc.pull(cv::gout(gapi_mat)) && num_frames < max_frames)
+    while (cc.pull(cv::gout(gapi_mat, seq_id, timestamp)) && num_frames < max_frames)
     {
         num_frames++;
-        cap >> ocv_mat;
-
-        cv::Mat y, uv;
-        cvtBGR2NV12(ocv_mat, y, uv);
-        cv::cvtColorTwoPlane(y, uv, ocv_mat, cv::COLOR_YUV2BGR_NV12);
 
+        cap >> cap_mat;
+        req_mat = fromBGR(cap_mat);
+        cv::resize(req_mat, ocv_mat, cv::Size(1920, 1080));
         EXPECT_EQ(0, cvtest::norm(ocv_mat, gapi_mat, NORM_INF));
+
+        all_seq_ids.push_back(seq_id);
+        all_timestamps.push_back(timestamp);
     }
+
+    cc.stop();
+
+    EXPECT_EQ(0L, all_seq_ids.front());
+    EXPECT_EQ(int64_t(max_frames) - 1L, all_seq_ids.back());
+    EXPECT_EQ(max_frames, all_seq_ids.size());
+    EXPECT_TRUE(std::is_sorted(all_seq_ids.begin(), all_seq_ids.end()));
+    EXPECT_EQ(all_seq_ids.size(), std::set<int64_t>(all_seq_ids.begin(), all_seq_ids.end()).size());
```

Comment:
`max_frames` vs `all_seq_ids` (which is already `set`)

Reply:
done